client/webpack.config.ts

```diff
@@ -18,6 +18,13 @@ module.exports = {
     client: {
       overlay: false,
       progress: true
+    },
+    // need to be configured via web server like nginx or apache in production mode
+    proxy: {
+      '/api/v1': {
+        target: 'http://localhost:3100',
+        router: () => 'http://localhost:3002/polymorpher.substack.com'
```

Comment:
shouldn't just point to a fixed substack site since we need to test many configurations in development mode. You can use router and parse the req, but I think you could just use rewrite - it could be a lot simpler

Reply:
Request `/api/v1` is made automatically from substack script to the default domain the script is running from. For example, on `https://polymorpher.substack.com`, request to `/api/v1` (hence https://polymorpher.substack.com/api/v1) is made. On the other hand, when a substack page is rendered in our client app `sample.com`, the api is made to `sample.com/api/v1` .

And we can't control anything about the request `/api/v1` because it is called from substack script, which means we can't add any parameter specifying which substack it is for to the request url from the client app.

Hence, in order to get what substack this request is about, I think we need to
- parse the request to get request origin
- query the smart contract with the request origin to get substack domain

from the backend code.
It feels pretty complicated code than testing it via webpack proxy, but want your confirmation before doing it.

Reply:
I understood all these. It was self-explanatory and doesn't require elaboration. What I was saying is we can control requests like /api/v1 sent to same origin (where the client is hosted) because we control the load balancer (see my other comment) and can rewrite the request in whatever way we need to, for example, to capture all requests sent to `/api/v1` and route them to a backend (proxy) server instead of the frontend server/bucket, and rewrite the path in URL if we need it. The effect is the same host and different paths are served by different servers. What I was saying above is in development environment (where you don't have a load balancer, unlike production environment), you can use webpack devserver's `pathRewrite`, see https://webpack.js.org/configuration/dev-server/#devserverproxy. This lets you proxy the requests to the local backend server (while preserving everything else, e.g. `host`). For example, set `target: https://localhost:8443` and `pathRewrite: { '^/api': '/coolproxy' }`, then all requests in the browser to`https://hello.localhost:3100/api/v1/blah` (where 3100 is serving the webpack client) will be routed to  `https://localhost:8443/coolproxy/v1/blah` but if you check `req.host` in the route handler of the backend server, you will get `hello.localhost:3100` - so essentially SLD is preserved and you can use it to query smart contrats. I also mentioned you can use router - I was referring to https://github.com/chimurai/http-proxy-middleware#router-objectfunction where you can use very custom logic instead of just a rewrite rule, but I think it is an overkill, like I mentioned before.

Reply:
I see. How about reducing the response time by removing smart contract query and hardcoding substack subdomain when configuring load balancer rewrite rules like
```
/api/v1 => backend-host/substack/polymorpher/api/v1
```
it will query directly to polymorpher.substack.com/api/v1 without querying substack subdomain to the smart contract

Reply:
We need to have the backend query smart contract based on SLD and subdomain because that's the whole point of doing all these - configure a SLD and its subdomain to connect to a user's substack of choice, not a substack we choose.

Reply:
Okay